src/nlp/metric.py

```diff
@@ -38,6 +38,17 @@
 logger = get_logger(__file__)
 
 
+class NoFileLock(BaseFileLock):
+    """Thread lock until file exists"""
+
+    def _acquire(self):
+        if os.path.exists(self._lock_file):
+            self._lock_file_fd = self._lock_file
```

Comment:
The thing is that the `_lock_file` persiste after the lock is release (it's not deleted) so you have to make a test that you cannot acquire the lock. Maybe the easiest is to do a wrapper around a FileLock object.

Reply:
Maybe I can replace `os.path.exists` with something that tries to open the file ?

Reply:
That's what the various platform-specific classes inherited from BaseFileLock do